### PR TITLE
Fixing bug on random seed initialization & data augmentation

### PR DESCRIPTION
**Pull request by RPetitpierre**
Issue ticket: https://github.com/dhlab-epfl/dhSegment-torch/issues/4
Solution: https://tanelp.github.io/posts/a-bug-that-plagues-thousands-of-open-source-ml-projects/

### CODE REVIEW

**Comment by raphaelBarman** [comment]
Maybe this should be changed in the `DataLoader` class and not in the `Trainer` class.
Moreover, it seems that the fix proposed is not correct (c.f. [discussion from author of np.random](https://github.com/pytorch/pytorch/issues/5059#issuecomment-817392562)).
C.f. [PyTorch-lightning fix](https://github.com/PyTorchLightning/pytorch-lightning/blob/851482165a9b5298d18b1aee8e555a5de105987f/pytorch_lightning/utilities/seed.py#L81) for a better correction.

**Comment by RPetitpierre** [comment]
Fixed following the recommendations found in [pytorch #5059 comment](https://github.com/pytorch/pytorch/issues/5059#issuecomment-817392562)

Epoch 0
[  69712850  138368793 3696970666   29957379]
[ 685742995 3720577954  554256087 2922606810]
[ 257418291   59715423  907653959 1746367900]
[2188036094  740733362 3103589195  974349332]
Epoch 1
[1477449950  234428354 1626646022  616016300]
[3790950317 2765397739 1593983494 1976647909] 
[2612116358 3241016252 3394135910  231797795]
[3757011889 3248610683 1904430329  748323930]
Epoch 2
[1414613552 1909377411 3908162682 4215379469]
[  42585860  808756280 2740704023 3264769525]
[3069659549 2004903103  828929030 2167140825]
[1869655976 3693980568  523287755 3537758458]
